lib/src/Pos/Node/API.hs

```diff
@@ -547,47 +553,209 @@ instance ToSchema (V1 Version) where
+    deriving (Show, Eq, Generic, ToJSON, FromJSON, Arbitrary)
+
+instance Buildable SecurityParameter where
+    build (SecurityParameter i) = bprint shown i
+
+instance ToSchema SecurityParameter where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "SecurityParameter") $ mempty
+            & type_ .~ SwaggerNumber
+            & maximum_ .~ Just (fromIntegral (maxBound :: Int))
+            & minimum_ .~ Just (fromIntegral (minBound :: Int))
+
+
+instance ToSchema (V1 Core.SlotId) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.SlotId") $ mempty
+            & type_ .~ SwaggerObject
+            & required .~ ["slot", "epoch"]
+            & properties .~ (mempty
+                & at "slot" ?~ (Inline $ mempty
+                    & type_ .~ SwaggerNumber
+                    & maximum_ .~ Just (fromIntegral (maxBound :: Word16))
+                    & minimum_ .~ Just (fromIntegral (minBound :: Word16))
+                    )
+                & at "epoch" ?~ (Inline $ mempty
+                    & type_ .~ SwaggerNumber
+                    & maximum_ .~ Just (fromIntegral (maxBound :: Word64))
+                    & minimum_ .~ Just (fromIntegral (minBound :: Word64))
+                    )
+                )
+
+instance ToJSON (V1 Core.SlotId) where
+    toJSON (V1 s) =
+        object
+            [ "epoch" .= toJSON (Core.getEpochIndex $ Core.siEpoch s)
+            , "slot"  .= toJSON (Core.getSlotIndex  $ Core.siSlot s)
+            ]
+
+instance FromJSON (V1 Core.SlotId) where
+    parseJSON = withObject "SlotId" $ \sl ->
+        Core.SlotId
+            <$> (fromInteger <$> sl .: "epoch")
+            <*> (Core.UnsafeLocalSlotIndex <$> sl .: "slot")
+            <&> V1
+
+instance Arbitrary (V1 Core.SlotId) where
+    arbitrary = fmap V1 arbitrary
+
+
+instance ToSchema Byte where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Byte") $ mempty
+            & type_ .~ SwaggerNumber
+
+instance Buildable Byte where
+    build b = bprint shown b
+
+
+instance Arbitrary (V1 Core.TxFeePolicy) where
+    arbitrary = fmap V1 arbitrary
+
+
+instance ToJSON (V1 Core.TxFeePolicy) where
+    toJSON (V1 p) =
+        object $ case p of
+            Core.TxFeePolicyTxSizeLinear (Core.TxSizeLinear a b) ->
+                [ "tag" .= ("linear" :: String)
+                , "a" .= toJSON a
+                , "b" .= toJSON b
+                ]
+            Core.TxFeePolicyUnknown policyTag policyPayload ->
+                [ "tag" .= ("unknown" :: String)
+                , "unknownTag" .= policyTag
+                , "unknownPayload" .= (BC8.unpack $ B64.encode policyPayload)
+                ]
+
+instance FromJSON (V1 Core.TxFeePolicy) where
+    parseJSON j = V1 <$> (withObject "TxFeePolicy" $ \o -> do
+        (tag :: String) <- o .: "tag"
+        case tag of
+            "linear" ->
+                Core.TxFeePolicyTxSizeLinear <$> parseJSON j
+            "unknown" -> do
```

Comment:
*If* possible there would be an opportunity here, I think, as we're already defining our own API specific types (currently `V1 Core.TxFeePolicy`), different from both `Core.TxFeePolicy` and the JSON representation.

Reply:
I think that makes total sense. Having an API representation that we map from and to `TxFeePolicy` :+1:

Reply:
So generics + specific types? I should have asked about this, sorry — thought we would have needed something much more aircastel-y, like DerivingVia, or something. I decided to stick with the handwritten instances to avoid unpleasant surprises for the time being.

I'm still not fully convinced there aren't any unpleasant surprises if we were to use generics instead. Are there?

My quick take:
- We need mapping both ways / isomorphism between `Core.SomeType` and `Node.API.SomeType`
- We would need to duplicate hierarchies of types [Core.TxFeePolicy](https://github.com/input-output-hk/cardano-sl/blob/develop/core/src/Pos/Core/Common/TxFeePolicy.hs#L44) so that the two `Coeff` can be replaced with `MeasuredIn 'AdaPerByte Coeff` and `MeasuredIn 'Ada Coeff`.
- We need Aeson + ToSchema instances for `MeasuredIn` (that's basically possible I believe?) Though might lose some flexibility (we can't provide the name in `Aeson.withObject "name"`)

Reply:
Might explore
```haskell
instance (ToJSON q, UnitToText u) => ToJSON (MeasuredIn (u :: UnitOfMeasure) q)`
```
instead of current helper-functions, in a separate PR.